daemon/config/config.go

```diff
@@ -600,6 +609,61 @@ func findConfigurationConflicts(config map[string]interface{}, flags *pflag.Flag
 	return nil
 }
 
+// MergeSpecialConfig verifies configs and handles conflicts in changed values
+func MergeSpecialConfig(config *Config, configFile string, flags *pflag.FlagSet) (*Config, error) {
+
+	unmergedConfig, _err := New()
+	if _err != nil {
+		return nil, _err
+	}
+
+	b, err := os.ReadFile(configFile)
+	if err != nil {
+		b = []byte("")
+	}
+	err = json.Unmarshal(b, unmergedConfig)
+
+	if err == nil && flags != nil {
+
+		mtu, _err := flags.GetInt("mtu")
+		if _err == nil && flags.Changed("mtu") {
```

Comment:
All these errors should cause a return (though admittedly, I'm not sure why `flags` would error here, please check).